docs/source/tutorial/dags.rst

```diff
@@ -0,0 +1,283 @@
+***********************
+Directed Acyclic Graphs
+***********************
+
+This tutorial will explore using retworkx to work with directed acyclic graphs
+(also known as a dag).
+
+Directed Graph
+==============
+
+A directed graph is a graph that is made up of a set of nodes connected by
+directed edges (often called arcs). Edges have a directionality which is
+different from undirected graphs where edges have no notion of a direction to
+them. For example:
+
+.. jupyter-execute::
+
+    import retworkx as rx
+    from retworkx.visualization import mpl_draw
+
+    path_graph = rx.generators.directed_path_graph(5)
+    mpl_draw(path_graph)
+
+In this example we created a 5 node directed path graph. This shows the
+directionality of the edges in the graph with the arrow head pointing to the
+target node. Each edge has a source and target
+
+Directed Acyclic Graphs
+=======================
+
+A directed acyclic graph is a directed graph which also doesn't contain
+any cycles. A cycle is a non-empty trail in which the first and last nodes in
+the trail are the same. For example:
+
+.. jupyter-execute::
+
+   cycle_graph = rx.generators.directed_cycle_graph(5)
+   mpl_draw(cycle_graph)
+
+is **not** acyclic. While the earlier path graph is acyclic.
+
+In retworkx you can create a new :class:`.PyDiGraph` object that enforces
+that no cycle is added by using the ``check_cycle`` constructor property
+
+.. jupyter-execute::
+
+    dag = rx.PyDiGraph(check_cycle=True)
+
+or after a :class:`~retworkx.PyDiGraph` object is created with the
+:attr:`~retworkx.PyDiGraph.check_cycle` attribute:
+
+.. jupyter-execute::
+
+    dag.check_cycle = True
+
+You can also check the status of cycle checking by inspecting the
+:attr:`~retworkx.PyDiGraph.check_cycle` attribute:
+
+.. jupyter-execute::
+
+    print(dag.check_cycle)
+
+With ``check_cycle`` set to ``True`` whenever a
+:meth:`.PyDiGraph.add_edge` or related functions that add an arbitrary edge to
+the graph an error will be raised if a cycle would be introduced. This checking
+does come with a noticeable (and potentially large) runtime overhead so it's
+best to only use it if it's strictly necessary. You can also avoid this
+overhead by using the :meth:`.PyDiGraph.add_parent` and
+:meth:`.PyDiGraph.add_child` as both add a new node and edge simultaneously and
+neither can introduce a cycle.
+
+
+Applications of DAGs
+====================
+
+Topological Sorting
+-------------------
+
+A toplogical sort of a directed acyclic graph defined as :math:`G = (V,E)` is
```

Comment:
typo: topological

Reply:
Fixed in: https://github.com/Qiskit/retworkx/pull/537/commits/f7e4421111f967d25a0b2303bb34e5e110fdd8cc